hibernate-core/src/main/java/org/hibernate/query/sqm/internal/ConcreteSqmSelectQueryPlan.java

```diff
@@ -427,6 +441,10 @@ private static CacheableSqmInterpretation buildCacheableSqmInterpretation(
 		final Map<QueryParameterImplementor<?>, Map<SqmParameter<?>, List<JdbcParametersList>>> jdbcParamsXref
 				= SqmUtil.generateJdbcParamsXref( domainParameterXref, sqmInterpretation::getJdbcParamsBySqmParam );
 
+		if ( !jdbcParamsXref.isEmpty() ) {
```

Comment:
I don't know if the performance is significantly affected by this flush mode check change, but if possible, I think I'd prefer to only do this when there is an entity valued parameter.